bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -169,6 +169,23 @@ spec:
                   fi
                   sleep 2
                   done
+
+                  # Upgrade the db if required.
+                  DB_SCHEMA="/usr/share/ovn/ovn-nb.ovsschema"
+                  DB_SERVER="unix:/var/run/ovn/ovnnb_db.sock"
+                  schema_name=$(ovsdb-tool schema-name $DB_SCHEMA)
+                  db_version=$(ovsdb-client -t 10 get-schema-version "$DB_SERVER" "$schema_name")
+                  target_version=$(ovsdb-tool schema-version "$DB_SCHEMA")
+
+                  if ovsdb-tool compare-versions "$db_version" == "$target_version"; then
+                    :
+                  elif ovsdb-tool compare-versions "$db_version" ">" "$target_version"; then
+                      echo "Database $schema_name has newer schema version ($db_version) than our local schema ($target_version), destroy the DB"
+                      rm -rf /etc/ovn/ovnnb_db.db
```

Comment:
I wonder if this will have the same implications that we are discussing with wiping all the dbs in upstream where ovn-controller and ovn-northd need to also be reset. @abhat FYI

For this particular case, northd probably doesn't matter because all of the master nodes are going to be bounced. However, for ovn-controller and sb_db the nodes will reject database updates until they get restarted, which might be OK on worker nodes since those will be upgraded after master nodes. Still what about ovnkube-node pods on the masters. What if they have come up and connected to the previous db before masters are done? Do we need to reset those?

Reply:
This code is executed after the nbdb and sbdb containers/pods are started and ovsdb-servers are already running. After deleting the db, I think its better to restart the containers. Otherwise there would be wierd behavior.